articles/200_coexistence_and_migration.md

```diff
@@ -0,0 +1,247 @@
+
+For the migration two different categories of changes need to be considered separately:
+
+* Changes which can be applied to ROS 1 packages to either make it compatible with ROS 2 or minimize the necessary changes.
+* Changes which are necessary to make the package work for ROS 2 but which are not compatible with ROS 1.
+
+If the second set of changes would be empty a package could be used as a ROS 1 package as well as ROS 2 packages using the same sources.
+Currently the previous sections mention a few different items which fall into the second category therefore it doesn't seem realistic to achieve compatibility using the exact same sources.
+The following sections assume that is not feasible to maintain a single source for the ROS 1 as well as the ROS 2 version of a package.
+
+### Changes applicable to ROS 1 packages
+
+* Update the package manifest to format 2.
+
+* In C++ introduce typedefs for all message / service classes as well as their pointer types.
+  This will allow to easily change the typedef to a different type in ROS 2.
+
+* Update Python code to also target Python 3.5+.
+
+### Changes necessary for migrating a package to ROS 2
+
+* Update the dependencies in the manifest.
+
+* Update the CMake code to find the updated dependencies.
+
+* The API of the message / service classes has changed:
+
+  * Different headers / modules to include / import.
+  * Different classes / methods / types.
+  * C++11 interface using e.g. `std::shared_ptr` rather then relying on `Boost`.
+
+* The API of the ROS client libraries has changed (`roscpp` -> `rclcpp`, `rospy` -> `rclpy`):
+
+  * Different headers / modules to include / import.
+  * Different classes / methods / types.
+
+## Maintenance
+
+When considering the migration of a package to ROS 2 a few things should be considered:
+
+* Is it planned to support a ROS 1 as well as a ROS 2 version of the package for longer time?
+
+  If not, the following options will likely matter less since only one version is maintained for longer.
+
+If yes, these two variations of the packages can be managed in different ways:
+
+* Both versions can be stored on separate branches within the same repository.
+
+  * (+) The code branches are closely together under the same "roof".
+  * (+) Changes between the two branches can be easily ported both ways.
+  * (-) Likely less discoverable for users.
+  * (-) It might be unclear which version reported issues are referring to.
+
+* One of the versions is stored in a forked repository.
+
+  * (-) The separation makes the packages appear less correlated.
+  * (+) Changes between the two branches can be easily ported both ways.
+  * (+) More obvious for users (assuming there is comparable infrastructure like distributions file, wiki, etc.).
+  * (+) It might be unclear which version reported issues are referring to.
+
+* Both versions are stored in separate unrelated repositories.
+
+  * No obvious advantages over the other twp approaches.
```

Comment:
typo: two

Reply:
Thanks. Fixed in c908ff0a18d10be6b5d979a37684a92c4eaf1966.